lib/pengine/utils.c

```diff
@@ -2392,9 +2392,19 @@ pe__clear_resource_history(pe_resource_t *rsc, pe_node_t *node,
 bool
 pe__rsc_running_on_any_node_in_list(pe_resource_t *rsc, GList *node_list)
 {
-    for (GList *ele = rsc->running_on; ele; ele = ele->next) {
-        pe_node_t *node = (pe_node_t *) ele->data;
-        if (pcmk__str_in_list(node_list, node->details->uname)) {
+    GList *hosts = NULL;
+
+    if (rsc == NULL || node_list == NULL) {
+        return false;
+    }
+
+    rsc->fns->location(rsc, &hosts, TRUE);
+
+    for (GList *hIter = hosts; hIter != NULL; hIter = hIter->next) {
+        pe_node_t *node = (pe_node_t *) hIter->data;
+
+        if (pcmk__str_in_list(node_list, node->details->uname) ||
+            pcmk__str_in_list(node_list, node->details->id)) {
```

Comment:
I just remembered we should do case-insensitive comparisons for node names

Reply:
Fixed.